### PR TITLE
Performance improvement

### PR DESCRIPTION
**Pull request by DaKnig**
this branch will track whatever performance improvements I can come up with and depends on #14 .
I will only add here changes that provably improve the performance in the tests available to me.
I will also later add the tests in a separate branch, including the setup to check performance.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -185,9 +185,12 @@ impl TestConfig {
 			result: None,
 		}
 	}
+	fn set_name(&mut self, name: String) {
```
Comment (evie-calico) [inline src/main.rs:188]: What is this function for? I'm not a fan of setter/getters that don't do anything

**Review by evie-calico — CHANGES_REQUESTED** [review]
This is great, thank you :)
Let me know if there's more you wanna add to this PR, otherwise I can merge this.

**Comment by DaKnig** [comment]
this is an ongoing PR I think I have a lot to contribute in terms of performance stuff :) I am trying to understand the characteristics of the performance. so far it seems like parallelizing does not add any speed. I have had a local version that did that and really even on huge tests the bottleneck is the test file parsing. there must be better ways to improve the speed before resorting to parallelism even if it is quite easy to implement.

**Comment by DaKnig** [comment]
I will add some changes that add tests, some performance metrics :)

File: src/main.rs (hunk 1 above)
Comment (DaKnig) [reply src/main.rs:188]: that's used in line 304. I want to change the whole approach from "do it with the global test, then the local one" to only have normal test configs and initialize them with the global config, but then I need to set the new name of course :) I realized an issue with my current approach and I think the whole TOML reading routine should be replaced. it is all WIP.

**Comment by evie-calico** [comment]
Could you rebase this? A lot of the changes seem out of date (copies of your first PR?) and some new stuff has been added.

**Comment by DaKnig** [comment]
the changes in this branch were merged already... would have been easier to track if merge policy was a bit different ;-)

**Pull request closed without merging**